web/src/types.ts

```diff
@@ -6,19 +6,111 @@ export type MatchedMessage = {
 
 // TODO/typescript: Move this to message_store
 export type MessageType = "private" | "stream";
+export type MessageContentType = "text/html" | "text/x-markdown";
+export type MessageReactionType = "unicode_emoji" | "realm_emoji" | "zulip_extra_emoji";
+
+// TODO/typescript: Move this to server_events
+export type TopicLink = {
+    text: string;
+    url: string;
+};
+
+// TODO/typescript: Move these types to message_store
+
+export type DisplayRecipientUser = {
+    email: string;
+    full_name: string;
+    id: number;
+    is_mirror_dummy: boolean;
+    unknown_local_echo_user?: boolean;
+};
+
+export type DisplayRecipient = string | DisplayRecipientUser[];
+
+export type MessageEditHistoryEntry = {
+    user_id: number | null;
+    timestamp: number;
+    prev_content?: string;
+    prev_rendered_content?: string;
+    prev_rendered_content_version?: number;
+    prev_stream?: number;
+    prev_topic?: string;
+    stream?: number;
+    topic?: string;
+};
+
+export type MessageReaction = {
+    emoji_name: string;
+    emoji_code: string;
+    reaction_type: MessageReactionType;
+    user_id: number;
+    user: DisplayRecipientUser;
+};
 
 // TODO/typescript: Move this to message_store
 export type RawMessage = {
```

Comment:
We can use this type in the future migration of the `message_helper`, `message_store` family of modules.
Marking the transition from `RawMessage` to `Message` can be tricky, because the mutation might happen in various places. `message_helper.process_new_message` may or may not get a `RawMessage`; the boolean-related fields are added in `message_store.set_message_booleans`; most of the functions in `reactions` calls `reactions.set_clean_reactions` to add `message.clean_reactions`. We need to determine where these transitions might happen to ensure that the types are correct. For this PR, I'm verifying if the functions called with `Message` in `people` indeed get a final `Message` object with all the fields added.